File: Deal/src/main/java/ru/leonov/deal/model/record/ApplicationHistoryElementRecord.java
package ru.leonov.deal.model.record;

import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import ru.leonov.deal.model.entity.ApplicationEntity;

import java.time.LocalDate;

/**
 * POJO that represent element of credit application history.
 */
@Data
@Getter
@Setter
public class ApplicationHistoryElementRecord {

    /**
     * New status of credit application.
     */
    private ApplicationEntity.Status status;

    /**
     * Date of status changing.
     */
    private LocalDate date;
}

File: Deal/deal_impl/src/main/kotlin/ru/leonov/deal/model/record/ApplicationHistoryElementRecord.kt
package ru.leonov.deal.model.record

import ru.leonov.deal.model.enums.ApplicationStatusEnum
import java.time.LocalDate

/**
 * POJO that represent element of credit application history.
 */
data class ApplicationHistoryElementRecord(
    /**
     * New status of credit application.
     */
    val status: ApplicationStatusEnum,
    /**
     * Date of status changing.
     */
    val date: LocalDate
)
